### PR TITLE
Use XDG_DATA_DIRS to query themes

### PR DESCRIPTION
**Pull request by Libadoxon**
Fixes [26](https://github.com/hyprwm/hyprcursor/issues/26). This also eliminates the need for a `NixOs` patch as it uses `XDG_DATA_DIRS` to link to all important directories. My C++ is also very bad so I hope I didn't screw anything up. It worked for me but other than that I didn't do any tests.

### CODE REVIEW

File: libhyprcursor/hyprcursor.cpp [hunk 1]
```diff
@@ -15,12 +16,25 @@
...
+static std::vector<std::string> getSystemThemeDirs() {
+    char*                    envXdgData = std::getenv("XDG_DATA_DIRS");
+    std::vector<std::string> result;
+    if (envXdgData) {
+        std::stringstream envXdgStream(envXdgData);
+        std::string       tmpStr;
+        while (getline(envXdgStream, tmpStr, ':')) {
+            result.push_back((tmpStr + "/icons"));
+        }
+    } else {
```
Comment (vaxerski) [inline libhyprcursor/hyprcursor.cpp:28]: no {} around short ifs

File: libhyprcursor/hyprcursor.cpp [hunk 2]
```diff
@@ -15,12 +16,25 @@
 
 using namespace Hyprcursor;
 
-// directories for lookup
-constexpr const std::array<const char*, 1> systemThemeDirs = {"/usr/share/icons"};
+static std::vector<std::string> getSystemThemeDirs() {
+    char*                    envXdgData = std::getenv("XDG_DATA_DIRS");
```
Comment (vaxerski) [inline libhyprcursor/hyprcursor.cpp:20]: can be const auto

File: libhyprcursor/hyprcursor.cpp [hunk 3]
```diff
@@ -15,12 +16,25 @@
...
+    std::vector<std::string> result;
+    if (envXdgData) {
+        std::stringstream envXdgStream(envXdgData);
+        std::string       tmpStr;
+        while (getline(envXdgStream, tmpStr, ':')) {
+            result.push_back((tmpStr + "/icons"));
+        }
+    } else {
+        std::vector<std::string> result = {"/usr/share/icons"};
```
Comment (vaxerski) [inline libhyprcursor/hyprcursor.cpp:29]: redefinition of result, did you mean `result = ...`?

File: libhyprcursor/hyprcursor.cpp [hunk 4]
```diff
@@ -15,12 +16,24 @@
...
+    } else
+        result = {"/usr/share/icons"};
+
+    return result;
+}
+
+const std::vector<std::string>             systemThemeDirs = getSystemThemeDirs();
 constexpr const std::array<const char*, 2> userThemeDirs   = {"/.local/share/icons", "/.icons"};
 
-//
```
Comment (vaxerski) [inline libhyprcursor/hyprcursor.cpp:22]: dont remove this

**Review by vaxerski — COMMENTED** [review]
rest lgtm also @fufexan for nix

**Review by fufexan — APPROVED** [review]
LGTM

**Review by vaxerski — APPROVED** [review]
thanks :)

**Pull request merged**